consumer/migration/hermes_migrator.go

```diff
@@ -85,65 +95,65 @@ func (m *HermesMigrator) Start(id string) error {
 		return nil
 	}
 
+	// if user not registered - do not do migration at all
 	registered, err := m.isRegistered(chainID, id)
 	if err != nil {
 		return fmt.Errorf("could not get identity register status: %w", err)
 	} else if !registered {
 		return errors.New("identity is unregistered")
 	}
 
-	activeHermes, err := m.addressProvider.GetActiveHermes(chainID)
-	if err != nil {
-		return fmt.Errorf("could not get hermes address: %w", err)
-	}
-	knownHermeses, err := m.addressProvider.GetKnownHermeses(chainID)
+	// get old and new hermeses
+	activeHermes, oldHermes, err := m.getHermeses(chainID)
 	if err != nil {
-		return fmt.Errorf("could not get hermes address: %w", err)
-	}
-	oldHermeses := getOldHermeses(knownHermeses, activeHermes)
-	if len(oldHermeses) != 1 {
-		log.Info().Msg("Migration interupted: there are more than 1 Hermes, cannot know which is which.")
-		return nil
+		return err
 	}
-	oldHermes := oldHermeses[0]
 
+	// get registry address
 	registryAddress, err := m.addressProvider.GetRegistryAddress(chainID)
 	if err != nil {
 		return fmt.Errorf("could not get registry address: %w", err)
 	}
+
+	// try open channel only if it's not opened yet.
+	if err = m.openChannel(id, err, chainID, activeHermes, registryAddress); err != nil {
+		return fmt.Errorf("open channel error: %w", err)
+	}
+
+	// get data from old hermes
 	data, err := m.getUserData(chainID, oldHermes.Hex(), id)
 	if err != nil {
```

Comment:
I think we can do the same here, if we have money or a promise on a new hermes and old hermes return no data, we can assume we're ok and migration was done.